deploy_docs.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+set -e # Exit with nonzero exit code if anything fails
+
+# Build the documentation from the SOURCE_BRANCH
+# and push it to TARGET_BRANCH.
+SOURCE_BRANCH="master"
+TARGET_BRANCH="gh-pages"
+
+# Pull requests and commits to other branches shouldn't try to deploy
+#if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_BRANCH" != "$SOURCE_BRANCH" ]; then
+#    echo "Skipping deploy on $TRAVIS_BRANCH. We only deploy docs automatically from development."
+#    exit 0
+#fi
+
+# Save some useful information
+REPO=`git config remote.origin.url`
+SSH_REPO=${REPO/https:\/\/github.com\//git@github.com:}
+SHA=`git rev-parse --verify HEAD`
+
+# Add rsa keys to the ssh agent to push to GitHub
+#gpg --output ../id_tardis_rsa --batch --passphrase $DECRYPT_GITHUB_AUTH --decrypt id_tardis_rsa.enc
+#chmod 600 ../id_tardis_rsa
+#if [ -z "$SSH_AUTH_SOCK" ] ; then
+#  eval `ssh-agent -s`
+#  ssh-add ~/.ssh/id_azure_rsa
+#fi
+
+
+# Clone the existing gh-pages for this repo into out/
+# Create a new empty branch if gh-pages doesn't exist yet (should only happen on first deply)
+git clone $REPO out
+cd out
+
+# # Regenerate the API documentation
+# git checkout $SOURCE_BRANCH || git checkout --orphan $SOURCE_BRANCH
+# sphinx-apidoc -f -M -o docs/source/ .
+
+# # Add new docs to the repo
+# git add docs/source --all
+
+# # Commit and push to SOURCE_BRANCH
+# git commit -m "Regenerate API Documentation: ${SHA}" || true
+# git push $SSH_REPO $SOURCE_BRANCH || true
+
+git checkout $TARGET_BRANCH || git checkout --orphan $TARGET_BRANCH
```

Comment:
what does `orphan` do?

Reply:
Orphan creates a new branch. This was originally Alice's code. So I suspect it is a fail safe, just incase you do not have gh-pages locally as a branch?

Reply:
A temporary new branch

Reply:
https://git-scm.com/docs/git-checkout#Documentation/git-checkout.txt---orphanltnewbranchgt

It creates a new branch without any history. I believe the idea is to help keep the `gh-pages` branch clean so that it only contains commits related to docs deployment.